core/shared/src/main/scala/spire/algebra/Field.scala

```diff
@@ -1,21 +1,61 @@
 package spire
 package algebra
 
+/** We precise AlgebraField[_] = algebra.Field[_] by extending GCDRing/EuclideanRing with
```

Comment:
OK, it means that algebra.Field only extends CRing, and algebra does not define the intermediate types.

We define our own Field to have the inheritance, and provide good defaults for gcd/lcm/quot ...